cheroot/workers/threadpool.py

```diff
@@ -151,12 +151,34 @@ def __init__(
             server (cheroot.server.HTTPServer): web server object
                 receiving this request
             min (int): minimum number of worker threads
-            max (int): maximum number of worker threads
+            max (int): maximum number of worker threads (-1/inf for no max)
             accepted_queue_size (int): maximum number of active
                 requests in queue
             accepted_queue_timeout (int): timeout for putting request
                 into queue
+
+        :raises ValueError: if the min/max values are invalid
+        :raises TypeError: if the max is not an integer or inf
         """
+        if min < 1:
+            raise ValueError('min={} must be > 0'.format(min))
```

Comment:
Now that I've allowed f-strings in the linter, it's possible to get bet to them. I'm not against `str.format()` but the placeholders should be named for it to be readable. Just `{}` is rather confusing.
```suggestion
            raise ValueError(f'min={min!s} must be > 0')
```